rskj-core/src/test/java/co/rsk/peg/BridgeSupportFlyoverTest.java

```diff
@@ -646,6 +649,168 @@ private BigInteger sendFundsToAnyAddress(
         );
     }
 
+    private BigInteger sendFundsSurpassesLockingCapToAnyAddress(
+        BtcTransactionProvider btcTransactionProvider,
+        boolean shouldTransferToContract,
+        Coin lockingCap
+    ) throws BlockStoreException, BridgeIllegalArgumentException, IOException {
+        when(activations.isActive(ConsensusRule.RSKIP134)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP176)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+
+        BridgeConstants bridgeConstants = spy(bridgeConstantsRegtest);
+
+        Context btcContext = mock(Context.class);
+        doReturn(bridgeConstants.getBtcParams()).when(btcContext).getParams();
+
+        Federation activeFederation = PegTestUtils.createFederation(bridgeConstants, "fa03", "fa04");
+        Federation retiringFederation = PegTestUtils.createFederation(bridgeConstants, "fa01", "fa02");
+
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+        when(provider.getNewFederation()).thenReturn(activeFederation);
+        when(provider.getOldFederation()).thenReturn(retiringFederation);
+        ReleaseTransactionSet releaseTransactionSet = new ReleaseTransactionSet(new HashSet<>());
+        when(provider.getReleaseTransactionSet()).thenReturn(releaseTransactionSet);
+
+        Address userRefundBtcAddress = PegTestUtils.createRandomBtcAddress(bridgeConstants.getBtcParams());
+        Address lpBtcAddress = PegTestUtils.createRandomBtcAddress(bridgeConstants.getBtcParams());
+
+        Repository repository = spy(createRepository());
+        // Set current bridge balance in order to create the condition to surpass the locking cap when needed
+        doReturn(co.rsk.core.Coin.valueOf(1)).when(repository).getBalance(any());
```

Comment:
If we are doing this then it doesn't really make much sense to receive the locking cap value as a parameter, right?

I feel it would much clearer if we set a given locking cap value (as it is now) and then simply try lo peg-in a value above it

Reply:
I agree. I got bias from the legacy code and didn't notice it can be done in that way. I already changed it. Take a look and let me know what you think.